### DISCUSSION TITLE
Recommended method to display vips image in Windows (c api only)

### DISCUSSION THREAD

**Discussion by sjards** [libvips/libvips#2933] [Help wanted]
Hello,

Can anyone please advise on an efficient method to display vips image data in Windows OS?
I'm currently using libvips with Autoit so limited to c-api calls or COM.

Currently I'm converting the image data to 32bit DIB format and displaying with GDI.
This works but I feel there must be a more efficient way.

I'm considering trying to implement SDL2, or Direct2d with WIC, but before I start down a route I thought I'd ask the experts first.

Many thanks

Steve

**Comment by jcupitt** [comment]
Hi @sjards,

Did you find vipsdisp-tiny? It's a very basic viewer in 300 lines of C:

https://github.com/libvips/vipsdisp-tiny/blob/master/vipsdisp.c

It has the nice feature of being asynchronous, so the UI is always live, and tiles are computed by a set of background threads as you scroll around.

I've done a more complete viewer here: https://github.com/jcupitt/vipsdisp

I've not tried to build on windows, though it ought to work. 

It works rather like a google maps style viewer. It keeps a tree of cached tiles on the GPU, where each tile has a position and a scale. As you pan and zoom, it finds the set of visible tiles and builds a scene graph that gets drawn to the display. Like vipsdisp-tiny, it uses `vips_sink_screen()` to compute any missing tiles in the background.

**Comment by sjards** [comment]
Many thanks @jcupitt :thumbsup:

I see vipsdisp-tiny is using gtk, I think I can work with that :smiley:
Thanks again for all the support you give here!
All the best
